src/into_elements.rs

```diff
@@ -0,0 +1,65 @@
+use data::{DataMap, Element, FromElements};
+use visit::{EdgeRef, Data, GraphBase,  IntoNodeIdentifiers, IntoEdgeReferences};
+use std::cell::RefCell;
+use std::collections::HashMap;
+use std::hash::Hash;
+use std::rc::Rc;
+
+/// Turn a graph with clonable weights that implementings the correct taits into its Elements.
+pub struct IntoElements<'a, G: 'a>(pub &'a G);
+
+
+//Todo: ids do not need to be cloneable. I couldn't get the lifetimes of EdgeWeight to work.
+/// Graphs that have a Cloneable weigths and implement the necessary traits can be deconstructed into elements.
+impl<'a, G> IntoIterator for IntoElements<'a, G>
+    where &'a G: GraphBase + DataMap + IntoNodeIdentifiers + IntoEdgeReferences,
+          <&'a G as Data>::NodeWeight: Clone,
+          <&'a G as Data>::EdgeWeight: Clone,
+          <&'a G as GraphBase>::NodeId: Eq + Hash + Clone,
+{
+    type Item = Element<<&'a G as Data>::NodeWeight, <&'a G as Data>::EdgeWeight>;
+    type IntoIter = Box<Iterator<Item = Self::Item> + 'a>;
+    fn into_iter(self) -> Self::IntoIter {
+        use std::iter;
+        let nodes = self.0.node_identifiers();
+        let edge_refs = self.0.edge_references();
+        let map = Rc::new(RefCell::new(HashMap::<<&'a G as GraphBase>::NodeId, usize>::new()));
```

Comment:
Why is there an Rc'd map here? Now I'm surprised.

Reply:
We'll find a way to resolve this without Rc's and `Box<Iterators>` before we can merge it.

Reply:
Feel free to rebase and rewrite commits while working on this, if it makes the commit history cleaner.